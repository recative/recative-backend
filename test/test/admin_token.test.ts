import { describe, expect, it } from "vitest";

import { RootFetcher } from "../src/fetcher/authFetcher";

describe("admin token crud", () => {
    const payload = {
        token: "38a44a3f-c108-412b-a910-9e47adc17c27",
    }

    it('create admin token', async () => {
        let date = new Date()
        // add 5 days
        date.setDate(date.getDate() + 5)
        const resp = await RootFetcher.path('/admin/token').method('post').create()({
            is_valid:true,
            comment:"a token for test",
            admin_permission: ["write"],
            token: payload.token,
            expired_at: date.toISOString(),
        })

        expect (resp.ok).toBe(true)
        expect (resp.data).matchSnapshot()
    });

    it('get admin token info', async () => {
        const resp = await RootFetcher.path('/admin/token/{token}').method('get').create()({
            token: payload.token,
        })

        expect (resp.ok).toBe(true)
        resp.data.expired_at = undefined
        expect (resp.data).matchSnapshot()
    });

    it('update admin token', async () => {
        const resp = await RootFetcher.path('/admin/token/{token}').method('put').create()({
            token: payload.token,
            is_valid: false,
            comment: "a token for test",
            admin_permission: ["read"],
            expired_at: undefined,
        })

        expect (resp.ok).toBe(true)
        expect (resp.data).matchSnapshot()

        const resp2 = await RootFetcher.path('/admin/token/{token}').method('get').create()({
            token: payload.token,
        })

        expect (resp2.ok).toBe(true)
        resp2.data.expired_at = undefined
        expect (resp2.data).matchSnapshot()
    })

    it('delete token', async () => {
        const resp = await RootFetcher.path('/admin/token/{token}').method('delete').create()({
            token: payload.token,
        })

        expect (resp.ok).toBe(true)
        expect (resp.data).matchSnapshot()
    });

    it('get all tokens', async () => {
        const resp = await RootFetcher.path('/admin/tokens').method('get').create()({})

        expect (resp.ok).toBe(true)
        const data = resp.data.map((item) => {
            item.expired_at = undefined
            return item
        })
        expect (data).matchSnapshot()
    })
});