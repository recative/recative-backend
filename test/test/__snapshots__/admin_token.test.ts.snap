// Vitest Snapshot v1

exports[`admin token crud > create admin token 1`] = `
{
  "token": "38a44a3f-c108-412b-a910-9e47adc17c27",
}
`;

exports[`admin token crud > delete token 1`] = `undefined`;

exports[`admin token crud > get admin token info 1`] = `
{
  "admin_permission": [
    "write",
  ],
  "comment": "a token for test",
  "expired_at": undefined,
  "is_valid": true,
  "token": "38a44a3f-c108-412b-a910-9e47adc17c27",
  "type": "admin",
}
`;

exports[`admin token crud > get all tokens 1`] = `
[
  {
    "admin_permission": [
      "read",
      "write",
    ],
    "comment": "test1",
    "expired_at": undefined,
    "is_valid": true,
    "token": "1919810",
    "type": "admin",
  },
  {
    "admin_permission": [
      "sudo",
    ],
    "comment": "test2",
    "expired_at": undefined,
    "is_valid": false,
    "token": "114514",
    "type": "admin",
  },
]
`;

exports[`admin token crud > update admin token 1`] = `undefined`;

exports[`admin token crud > update admin token 2`] = `
{
  "admin_permission": [
    "read",
  ],
  "comment": "a token for test",
  "expired_at": undefined,
  "is_valid": true,
  "token": "38a44a3f-c108-412b-a910-9e47adc17c27",
  "type": "admin",
}
`;
